### PR TITLE
Add BigONE V3 API

### PR DESCRIPTION
**Pull request by Roachard**
BigONE V2 API will be deprecated in Fri, 11 Oct 2019 04:00:00 GMT.  
I try to add the V3 API base on the old version js, and it seems that it works.
Some new features like 'fetchMyTrades' and 'fetchOHLCV' are not completed yet, but it should be fine as they're not supported by V2 API any way.

### CODE REVIEW

**Comment by blackstock** [comment]
@kroitor Any chance of this being merged soon?

**Comment by blackstock** [comment]
@kroitor just bumping this thread. BigOne is no longer supported as it's currently implemented. Any shot of a merge this week or do you think there is more work required on this PR? Happy to help if I can

**Comment by kroitor** [comment]
@blackstock i'll do my best to get to it and comment on the code thoughtfully as soon as I can, there are some minor issues on the surface, like min limits not populated in `fetchMarkets`, and some missing error handling for `InsufficientFunds` and order-related errors, a few violations of [CONTRIBUTING guidelines](https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#accessing-dictionary-keys) in particular [here](https://github.com/ccxt/ccxt/pull/5904/files#diff-0c276ec42415565819931c349b3b424eR386), also a potential [division by zero](https://github.com/ccxt/ccxt/pull/5904/files#diff-0c276ec42415565819931c349b3b424eR404), it is also missing the fix for [market buy orders](https://github.com/ccxt/ccxt/pull/6013), that is now available in the master in the exchanges having `options['createMarketBuyRequiresPrice']`, etc...

So, basically, it needs QA, however, those fixes would be one-liner edits mostly, and there's many of them throughout the file, so, it's often easier to just edit those lines by myself instead of commenting on the line just for someone to make the same quickfix (that would be redundant and a would just waste time). 

I'm sorry for the delays with my reaction recently – those are mostly caused by a lack of time, as we're working on [another important piece of the CCXT framework](https://github.com/ccxt/ccxt/issues/56#issuecomment-507290536) as you may know... We also have to find a way to add even more automation for the above, apparently our existing CI and `eslint` don't catch everything.

Your help is always very much appreciated!

**Comment by blackstock** [comment]
@kroitor I know you're swamped on the new framework feature but was curious if you had an ETA on this PR? The existing BigOne API has been broken since they sunset v2 of their API and I may need to hand-roll my own integration to grab data until this gets merged. 

Thanks again for all the hard work!

**Comment by kroitor** [comment]
@blackstock hope to get to it within a week or so.

**Comment by blackstock** [comment]
Just a friendly reminder :) I know you're swamped. Hope all is well @kroitor !

**Comment by kroitor** [comment]
@blackstock terribly sorry for the delays again, will do my best to merge it this week, we're close.
